shared/src/question-generators/math/eea.ts

```diff
@@ -0,0 +1,214 @@
+      })
+      .join("; ")
+
+    // check final coefficients for linear combination
+    const initialStep = steps[0]
+    const a = initialStep.dividend
+    const b = initialStep.divisor
+    const finalStep = steps[steps.length - 1]
+    const gcd = finalStep.remainder
+    const userCoefA = parseInt(text["coefA"], 10)
+    const userCoefB = parseInt(text["coefB"], 10)
+    const coefficientsCorrect = userCoefA === finalStep.s && userCoefB === finalStep.t
+
+    // correctness for steps and coefficients
+    const isCorrect = allStepsCorrect && coefficientsCorrect
+
+    return {
+      correct: isCorrect,
+      correctAnswer: isCorrect
+        ? ""
+        : `${feedbackDetails}\\n$ \\text{gcd}(${a}, ${b}) = ${gcd} = ${finalStep.s} \\cdot ${a} + ${finalStep.t} \\cdot ${b} $`,
+    }
+  }
+}
+
+function calculateEEA(a: number, b: number) {
+  const steps = []
+  let s0 = 1,
+    t0 = 0,
+    s1 = 0,
+    t1 = 1
+
+  while (b !== 0) {
+    const quotient = Math.floor(a / b)
+    const remainder = a % b
+    const sNext = s0 - quotient * s1
+    const tNext = t0 - quotient * t1
+
+    steps.push({
+      dividend: a,
+      divisor: b,
+      quotient,
+      remainder,
+      s: s0,
+      t: t0,
+    })
+
+    // update for next iteration
+    a = b
+    b = remainder
+    s0 = s1
+    t0 = t1
+    s1 = sNext
+    t1 = tNext
+  }
+
+  // gcd is last non-zero remainder (stored invisibly in a)
+  const gcd = a
+
+  // push final coefficients
+  steps.push({
+    dividend: a,
+    divisor: null, // no valid divisor
+    quotient: null, // no valid quotient
+    remainder: gcd, // gcd is remainder in final line
+    s: s0,
+    t: t0,
+  })
+
+  return { gcd, steps }
+}
+
+function generateEEATableSteps(
+  a: number,
+  b: number,
+  lang: Language,
+  steps: Array<{
+    dividend: number
+    divisor: number | null
+    quotient: number | null
+    remainder: number
+    s: number
+    t: number
+  }>,
+) {
+  const divisionSteps = steps
+    .filter((step, index) => index === 0 || step.quotient !== null)
+    .map((step, index) => {
+      const stepIndex = index + 1
+
+      // first row displays known quantities a and b
+      return index === 0
+        ? `| $ ${step.dividend} $ |$=$| {{quotient${stepIndex}#TL#}} |$\\cdot$| ${step.divisor} |$+$| {{remainder${stepIndex}#TL#}} |\n`
+        : `| {{dividend${stepIndex}#TL#}} |$=$| {{quotient${stepIndex}#TL#}} |$\\cdot$| {{divisor${stepIndex}#TL#}} |$+$| {{remainder${stepIndex}#TL#}} |\n`
+    })
+
+  const calcTable = [
+    //"\n| **Dividend** |$=$| **Quotient** |$\\cdot$| **Divisor** |$+$| **Remainder** |\n",
+    //"|-----------|--|---------|-----|---------|--|-----------|\n",
+    ...divisionSteps,
+  ].join("")
+
+  const linearCombinationPrompt = `\n${t(translations, lang, "linearCombinationPrompt")}\n`
+  const finalRow = `| $\\text{${t(translations, lang, "gcd")}}(${a},${b})$ |$=$| {{gcd#TL#}} |$=$| {{coefA#TL#}} |$\\cdot$| ${a} |$+$| {{coefB#TL#}} |$\\cdot$| ${b} |\n`
```

Comment:
Maybe split up into two rows, this is very compact on mobile? 
<img width="431" alt="Screenshot 2024-11-04 at 19 30 00" src="https://github.com/user-attachments/assets/e643e3be-05ee-4261-8d31-95ad8c0c98e1">